src/packages/engine-core/src/NodeEngine.ts

```diff
@@ -781,32 +802,63 @@ ${this.lastErrorLog.fields.file}:${this.lastErrorLog.fields.line}:${this.lastErr
         }
 
         this.url = `http://localhost:${this.port}`
+
+          // don't wait for this
+
+          ; (async () => {
+            const engineVersion = await this.version()
+            debug(`Client Version ${this.clientVersion}`)
+            debug(`Engine Version ${engineVersion}`)
+          })()
+
         resolve()
       } catch (e) {
         reject(e)
       }
     })
   }
 
+  async stop(): Promise<void> {
+    if (!this.stopPromise) {
+      this.stopPromise = this._stop()
+    }
+
+    return this.stopPromise
+  }
+
   /**
    * If Prisma runs, stop it
    */
-  async stop(): Promise<void> {
-    await this.start()
+  async _stop(): Promise<void> {
+    if (this.startPromise) {
+      await this.startPromise
+    }
+    await new Promise(r => process.nextTick(r))
     if (this.currentRequestPromise) {
       try {
         await this.currentRequestPromise
       } catch (e) {
         //
       }
     }
+    this.getConfigPromise = undefined
+    let stopPromise
     if (this.child) {
       debug(`Stopping Prisma engine`)
+      if (this.startPromise) {
+        await this.startPromise
+      }
+      stopPromise = new Promise((resolve, reject) => {
+        this.engineStopDeferred = { resolve, reject }
+      })
       this.queryEngineKilled = true
       this.undici?.close()
       this.child?.kill()
       delete this.child
     }
+    this.startPromise = undefined
+    await stopPromise
```

Comment:
When `this.child` is truthy that means that stopPromise is a Promise but if it's falsy it would be `undefined` meaning it does `await undefined` which is a bit weird but works I guess?

Reply:
Good point, added an `if(stopPromise)` in there!